stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channels/list/ChannelItem.kt

```diff
@@ -0,0 +1,246 @@
+package io.getstream.chat.android.compose.ui.channels.list
+
+import androidx.compose.foundation.ExperimentalFoundationApi
+import androidx.compose.foundation.background
+import androidx.compose.foundation.combinedClickable
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.material.Icon
+import androidx.compose.material.Text
+import androidx.compose.material.ripple.rememberRipple
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.AnnotatedString
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.compose.R
+import io.getstream.chat.android.compose.state.channel.list.ChannelItemState
+import io.getstream.chat.android.compose.ui.components.Timestamp
+import io.getstream.chat.android.compose.ui.components.avatar.ChannelAvatar
+import io.getstream.chat.android.compose.ui.components.channels.MessageReadStatusIcon
+import io.getstream.chat.android.compose.ui.components.channels.UnreadCountIndicator
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.ui.util.getLastMessage
+
+/**
+ * The basic channel item, that shows the channel in a list and exposes single and long click actions.
+ *
+ * @param channelItem The channel data to show.
+ * @param currentUser The user that's currently logged in.
+ * @param onChannelClick Handler for a single tap on an item.
+ * @param onChannelLongClick Handler for a long tap on an item.
+ * @param modifier Modifier for styling.
+ * @param leadingContent Customizable composable function that represents the leading content of a channel item, usually
+ * the avatar that holds an image of the channel or its members.
+ * @param detailsContent Customizable composable function that represents the center content of a channel item, usually
+ * holding information about its name and the last message.
+ * @param trailingContent Customizable composable function that represents the trailing content of the a channel item,
+ * usually information about the last message and the number of unread messages.
+ */
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+public fun DefaultChannelItem(
```

Comment:
Do we want to keep the naming "Default" prefix for a component that is public and exposes a lot of slots?

This component seems like something heavily user oriented.

Reply:
Definitely should be just `ChannelItem`